source/destructurable.d.ts

```diff
@@ -0,0 +1,50 @@
+import {type Simplify} from './simplify';
+import {type UnionToIntersection} from './union-to-intersection';
+
+// Helper type. Not useful on its own.
+type AllKeysWithUndefined<Type> = UnionToIntersection<{
```

Comment:
I think this should be documented and tested, according to: https://github.com/sindresorhus/type-fest/blob/9feb8c89be9a0f2f688bf2f497230298a8e2472e/.github/contributing.md?plain=1#L17